esp/esdllib/esdl_xpath_extensions_libxml.cpp

```diff
@@ -472,6 +474,326 @@ static void strTokenizeFunction(xmlXPathParserContextPtr ctxt, int nargs)
+                    return;
+                }
+            }
+        }
+        xmlFree(tmp);
+    }
+    if (1 <= nargs)
+    {
+        tmp = xmlXPathPopString(ctxt);
+        if (xmlXPathCheckError(ctxt))
+            return;
+        if (*tmp)
+        {
+            if (strieq((const char*)tmp, "json"))
+                style = TXSUMMARY_OUT_JSON;
+            else if (!strieq((const char*)tmp, "text"))
+            {
+                xmlFree(tmp);
+                xmlXPathSetError((ctxt), XPATH_EXPR_ERROR);
+                return;
+            }
+        }
+        xmlFree(tmp);
+    }
+
+    StringBuffer output;
+    txSummary->serialize(output, level, group, style);
+    xmlXPathReturnString(ctxt, xmlStrdup((const xmlChar*)output.str()));
+}
+
+static void deprecatedEncryptString(xmlXPathParserContextPtr ctxt, int nargs)
+{
+    if (nargs != 1)
+    {
+        xmlXPathSetArityError(ctxt);
+        return;
+    }
+
+    xmlChar *toEncrypt = xmlXPathPopString(ctxt);
+    if (xmlXPathCheckError(ctxt)) //includes null check
+        return;
+
+    StringBuffer encrypted;
+    try
+    {
+        encrypt(encrypted, (const char*)toEncrypt);
+    }
+    catch (IException* e)
+    {
+        e->Release();
+        xmlXPathSetError(ctxt, XPATH_INVALID_OPERAND);
+    }
+    catch (...)
+    {
+        xmlXPathSetError(ctxt, XPATH_INVALID_OPERAND);
+    }
+    if (!xmlXPathCheckError(ctxt))
+        xmlXPathReturnString(ctxt, xmlStrdup((const xmlChar *)encrypted.str()));
+    xmlFree(toEncrypt);
+}
+
+static void deprecatedDecryptString(xmlXPathParserContextPtr ctxt, int nargs)
+{
+    if (nargs != 1)
+    {
+        xmlXPathSetArityError(ctxt);
+        return;
+    }
+
+    xmlChar *toDecrypt = xmlXPathPopString(ctxt);
+    if (xmlXPathCheckError(ctxt)) //includes null check
+        return;
+
+    StringBuffer decrypted;
+    try
+    {
+        decrypt(decrypted, (const char*)toDecrypt);
+    }
+    catch (IException* e)
+    {
+        e->Release();
+        xmlXPathSetError(ctxt, XPATH_INVALID_OPERAND);
+    }
+    catch (...)
+    {
+        xmlXPathSetError(ctxt, XPATH_INVALID_OPERAND);
+    }
+    if (!xmlXPathCheckError(ctxt))
+        xmlXPathReturnString(ctxt, xmlStrdup((const xmlChar *)decrypted.str()));
+    xmlFree(toDecrypt);
+}
+
+static void encodeBase64String(xmlXPathParserContextPtr ctxt, int nargs)
+{
+    if (0 == nargs || nargs > 2)
+    {
+        xmlXPathSetArityError(ctxt);
+        return;
+    }
+
+    xmlChar *toEncode = xmlXPathPopString(ctxt);
+    if (xmlXPathCheckError(ctxt)) //includes null check
+        return;
+    bool lineBreaks = false;
+    if (2 == nargs)
+    {
+        lineBreaks = xmlXPathPopBoolean(ctxt);
+        if (xmlXPathCheckError(ctxt))
+        {
+            xmlFree(toEncode);
+            return;
+        }
+    }
+
+    StringBuffer encoded;
+    JBASE64_Encode((const char*)toEncode, long(strlen((const char*)toEncode)), encoded, lineBreaks);
```

Comment:
Even if we wait for decode (see below) you could make a case for still being able to base64 encode text.  Not sure it's totally useful, but it's safe.